staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1435,7 +1457,13 @@ func (e *Store) Watch(ctx context.Context, options *metainternalversion.ListOpti
 
 // WatchPredicate starts a watch for the items that matches.
 func (e *Store) WatchPredicate(ctx context.Context, p storage.SelectionPredicate, resourceVersion string, sendInitialEvents *bool) (watch.Interface, error) {
-	storageOpts := storage.ListOptions{ResourceVersion: resourceVersion, Predicate: p, Recursive: true, SendInitialEvents: sendInitialEvents}
+	storageOpts := storage.ListOptions{
+		ResourceVersion:    resourceVersion,
+		Predicate:          p,
+		Recursive:          true,
+		SendInitialEvents:  sendInitialEvents,
+		WatchWithoutPrevKV: utilfeature.DefaultFeatureGate.Enabled(features.WatchFromStorageWithoutPrevKV),
```

Comment:
This is incorrect place to set `WatchWithoutPrevKV`. It should only be used by cacher.